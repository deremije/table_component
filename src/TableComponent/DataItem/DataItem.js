import React from "react"

class DataItem extends React.Component {
    render() {
        return (
            <div className='dataItem'>
                {this.props.line.photo && this.props.header === "name" ? <img src={this.props.line.photo} alt={this.props.line.name} /> : ""}{this.props.line[this.props.header]}
            </div>
        )
    }
}

export default DataItem