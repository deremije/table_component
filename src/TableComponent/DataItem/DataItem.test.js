import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DataItem from './DataItem'

Enzyme.configure({ adapter: new Adapter() })

describe('DataItem', () => {
    it ('should render without crashing', () => {
        const line = {name: "jeremy", photo: "jeremy.jpg", city: "lyon"}
        const header = 'city'
        const wrapper = shallow(<DataItem line={line} header={header} />)
        expect(wrapper).toMatchSnapshot()
    })

    it ('renders text', () => {
        const line = {name: "jeremy", photo: "jeremy.jpg", city: "lyon"}
        const header = 'city'
        const wrapper = shallow(<DataItem line={line} header={header} />)
        expect(wrapper.find('.tC-dataItem').text()).toContain("lyon")
    })

    it ('shows an image in name column', () => {
        const line = {name: "jeremy", photo: "jeremy.jpg"}
        const header = 'name'
        const wrapper = shallow(<DataItem line={line} header={header} includePhoto={"name"}/>)
        expect(wrapper.find('.tC-image').length).toEqual(1)
    })

    it ('does not show an image in any other case', () => {
        const line = {name: "jeremy", photo: "jeremy.jpg"}
        const header = 'photo'
        const wrapper = shallow(<DataItem line={line} header={header} includePhoto={"name"}/>)
        expect(wrapper.find('.tC-image').length).toEqual(0)
    })
})