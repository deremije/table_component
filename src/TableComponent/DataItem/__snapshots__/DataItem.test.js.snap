// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataItem should render without crashing 1`] = `ShallowWrapper {}`;
